src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs

```diff
@@ -142,150 +140,186 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
         {
-            var names = new HashSet<string>(new[] { SummaryTagName, RemarksTagName, ExampleTagName, CompletionListTagName });
+            elementName = null;
 
-            RemoveExistingTags(parentTrivia, names, (x) => x.StartTag.Name.LocalName.ValueText);
+            if (token.IsKind(SyntaxKind.XmlTextLiteralToken) && string.IsNullOrWhiteSpace(token.Text))
+            {
+                // Unlike VB, the C# lexer has a preference for leading trivia. In the following example...
+                //
+                //    /// <exception          $$
+                //
+                // ...the trailing whitespace will not be attached as trivia to any node. Instead it will
+                // be treated as an independent XmlTextLiteralToken, so skip backwards by one token.
+                token = token.GetPreviousToken();
+            }
 
-            return names.Select(GetItem);
-        }
+            // Handle the <elem$$ case by going back one token (the subsequent checks need to account for this)
+            token = token.GetPreviousTokenIfTouchingWord(position);
 
-        private void RemoveExistingTags(DocumentationCommentTriviaSyntax parentTrivia, ISet<string> names, Func<XmlElementSyntax, string> selector)
-        {
-            if (parentTrivia != null)
+            SyntaxList<XmlAttributeSyntax> attributes = default(SyntaxList<XmlAttributeSyntax>);
+
+            if (token.IsKind(SyntaxKind.IdentifierToken) && token.IsParentKind(SyntaxKind.XmlName))
+            {
+                // <elem $$
+                // <elem attr$$
+                elementName = GetElementNameAndAttributes(token.Parent.Parent, out attributes);
+            }
+            else if (token.IsParentKind(SyntaxKind.XmlCrefAttribute) ||
+                     token.IsParentKind(SyntaxKind.XmlNameAttribute) ||
+                     token.IsParentKind(SyntaxKind.XmlTextAttribute))
             {
-                foreach (var node in parentTrivia.Content)
+                // <elem attr="" $$
+                // <elem attr="" $$attr	
+                // <elem attr="" attr$$
+                var attributeSyntax = (XmlAttributeSyntax)token.Parent;
+
+                if (token == attributeSyntax.EndQuoteToken)
                 {
-                    var element = node as XmlElementSyntax;
-                    if (element != null)
-                    {
-                        names.Remove(selector(element));
-                    }
+                    elementName = GetElementNameAndAttributes(attributeSyntax.Parent, out attributes);
                 }
             }
+
+            attributeNames = attributes.Select(GetAttributeName).ToSet();
+            return elementName != null;
         }
 
-        private IEnumerable<CompletionItem> GetTagsForSymbol(ISymbol symbol, DocumentationCommentTriviaSyntax trivia)
+        private string GetElementNameAndAttributes(SyntaxNode node, out SyntaxList<XmlAttributeSyntax> attributes)
```

Comment:
Aye, this was written long before tuples were available. Will add ... with some fancy deconstruction :)